### PR TITLE
FieldMatchers: Add match by value (reducer)

### PR DESCRIPTION
**Pull request by leeoniya**
supersedes https://github.com/grafana/grafana/pull/57657

closes/resolves a big feature parity request with Graph (old): https://github.com/grafana/grafana/discussions/38299

besides conditionall hiding of fields (all zeros, or all nulls) this lets us do cool stuff like omit series from a VizRepeat, or re-color a series based on Last, etc.

https://user-images.githubusercontent.com/43234/223892752-aeeebdc3-d6ae-4d34-aedc-9a5b965a3853.mp4

### CODE REVIEW

**Comment by ryantxu** [comment]
@leeoniya -- i pushed some big changes...

**Comment by leeoniya** [comment]
❤️ that was gonna be next :)

**Comment by torkelo** [comment]
Looks great! just missing some tests :)

File: packages/grafana-data/src/transformations/matchers/compareValues.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      return left > right;
+    case ComparisonOperation.GTE:
+      return left >= right;
+    case ComparisonOperation.LT:
+      return left < right;
+    case ComparisonOperation.LTE:
+      return left <= right;
+    default:
+      return false;
+  }
```
Comment (leeoniya) [inline packages/grafana-data/src/transformations/matchers/compareValues.ts:41]: :thinking: 

https://github.com/grafana/grafana/blob/8ef2afda870d16451545f5a686ace348d67f0665/packages/grafana-data/src/transformations/matchers/ids.ts#L43-L54

Comment (ryantxu) [reply packages/grafana-data/src/transformations/matchers/compareValues.ts:41]: 🤔

**Comment by leeoniya** [comment]
LGTM, i can't approve my own PR though ;)

**Review by ryantxu — APPROVED** [review]

**Comment by melroy89** [comment]
This fields overide doesn't work with InfluxDB. 

With a simple query like:

```
SELECT count("url") AS "sum" FROM "access_log" WHERE $timeFilter GROUP BY time($interval)
```

This `count()` will have zeros in some cases, since it's counting based on the `$interval` (eg. like 5 seconds).

I selected "Fields with values" -> "All zeros" -> Hide tooltip, hide VIz and Hide Legend.

This is using the latest Time series (new) graph.

However despite using "Fields with values" -> "All zeros" and trying to hide these 0 values from the graph, tool-tip and legend. It doesn't work at all...

image

**Comment by robinpecha** [comment]
The same problem with Clickhouse datasource.
image

**Comment by melroy89** [comment]
> The same problem with Clickhouse datasource. 

I see.. Let's create an official issue ticket out of it. And link to this PR? Shall we?

**Pull request merged**